apps/calendar/js/next_tick.js

```diff
@@ -1,29 +1,14 @@
 define(function(require, exports, module) {
 'use strict';
 
-var NEXT_TICK = 'calendar-next-tick';
-
-var nextTickStack = [];
+var resolved = Promise.resolve();
 
 /**
  * Very similar to node's nextTick.
- * Much faster then setTimeout.
+ * Much faster than setTimeout or window.postMessage
  */
 module.exports = function(callback) {
-  nextTickStack.push(callback);
-  window.postMessage(NEXT_TICK, '*');
+  resolved.then(callback);
```

Comment:
100% sure, it's over 100x faster: http://jsperf.com/test002/8 - I even talked to dbaron about it and also created a blog post: http://blog.millermedeiros.com/promise-nexttick/

this change improved the startup perf in ~80ms.